### DISCUSSION TITLE
Install Kubernetes Cluster Behind Tailscale VPN - Vincze Janos Blog

### DISCUSSION THREAD

**Discussion by giscus** [jvincze84/jvincze84.github.io#6] [General]
# Install Kubernetes Cluster Behind Tailscale VPN - Vincze Janos Blog

https://readthedocs.vinczejanos.info/Blog/2023/04/21/install-kube-tailscale/

**Comment by armaneshaghi** [comment]
Thanks for the great tutorial. I have followed it and got my cluster "almost" working. However, my internode connections managed by Weave-net fail because I think that the Weave is using its own Mac address to connect to the not (that is physically separate in a different subnet but of course is part of Tailnet). This causes a failure of connection in a relatively random and haphazard way (the pods sometimes, kubeadm jon works but running cluster eventually fails). 

An example log from my weave-net pod (on my control-plane node)

INFO: 2023/11/04 20:22:22.705133 ->[100.117.235.131:50895|06:e0:32:47:0b:bf(orcams)]: connection shutting down due to error: read tcp 100.80.247.73:6783->100.117.235.131:50895: read: connection reset by peer


The Mac address points to "weave" interface but not tailscale and perhaps (my assumption) is that this is the root cause of the problem. Should Weave be made aware that it needs to use Tailscale for internode communications?

**Reply by armaneshaghi** [reply]
For clarity: I have checked my Tailnet ACS (Access controls), and they do allow all connections. I also turned off my firewall on both of the nodes.
